app/HMS/User/Permissions/RoleManager.php

```diff
@@ -0,0 +1,177 @@
+<?php
+
+namespace HMS\User\Permissions;
+
+use HMS\Repositories\RoleRepository;
+use LaravelDoctrine\ORM\Facades\EntityManager;
+use LaravelDoctrine\ACL\Permissions\Permission;
+
+class RoleManager
+{
+    private $roleRepository;
+    private $permissionRepository;
+
+    /**
+     * Create a new RoleManager instance.
+     *
+     * @param HMS\Repositories\RoleRepository $roleRepository An instance of a role repository
+     */
+    public function __construct(RoleRepository $roleRepository)
+    {
+        $this->roleRepository = $roleRepository;
+        $this->permissionRepository = EntityManager::getRepository(Permission::class);
```

Comment:
looks like we would have to add it via the appServiceProvider and inject an ObjectRepository
not sure it thats going to obfuscate it away to much and if it used in various places we will never be 100% sure just which one is being injected